hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/create/CreateDecoder.java

```diff
@@ -0,0 +1,472 @@
+     * @return the synthetic transaction body
+     */
+    public TransactionBody decodeCreateNonFungibleWithCustomFeesV2(
+            @NonNull final byte[] encoded,
+            @NonNull final AccountID senderId,
+            @NonNull final HederaNativeOperations nativeOperations,
+            @NonNull final AddressIdConverter addressIdConverter) {
+        final var call = CreateTranslator.CREATE_NON_FUNGIBLE_TOKEN_WITH_CUSTOM_FEES_V2.decodeCall(encoded);
+        final TokenCreateWrapper tokenCreateWrapper = getTokenCreateWrapperNonFungibleWithCustomFees(
+                call.get(0), call.get(1), call.get(2), senderId, nativeOperations, addressIdConverter);
+        return bodyOf(createToken(tokenCreateWrapper));
+    }
+
+    /**
+     * Decodes a call to {@link CreateTranslator#CREATE_NON_FUNGIBLE_TOKEN_WITH_CUSTOM_FEES_V3} into a synthetic {@link TransactionBody}.
+     *
+     * @param encoded the encoded call
+     * @return the synthetic transaction body
+     */
+    public TransactionBody decodeCreateNonFungibleWithCustomFeesV3(
+            @NonNull final byte[] encoded,
+            @NonNull final AccountID senderId,
+            @NonNull final HederaNativeOperations nativeOperations,
+            @NonNull final AddressIdConverter addressIdConverter) {
+        final var call = CreateTranslator.CREATE_NON_FUNGIBLE_TOKEN_WITH_CUSTOM_FEES_V3.decodeCall(encoded);
+        final TokenCreateWrapper tokenCreateWrapper = getTokenCreateWrapperNonFungibleWithCustomFees(
+                call.get(0), call.get(1), call.get(2), senderId, nativeOperations, addressIdConverter);
+        return bodyOf(createToken(tokenCreateWrapper));
+    }
+
+    private TransactionBody bodyOf(@NonNull final TokenCreateTransactionBody.Builder tokenCreate) {
+        return TransactionBody.newBuilder().tokenCreation(tokenCreate).build();
+    }
+
+    private static TokenCreateWrapper getTokenCreateWrapper(
+            @NonNull final Tuple tokenCreateStruct,
+            final boolean isFungible,
+            final long initSupply,
+            final int decimals,
+            @NonNull final AccountID senderId,
+            @NonNull final HederaNativeOperations nativeOperations,
+            @NonNull final AddressIdConverter addressIdConverter) {
+
+        final var tokenName = (String) tokenCreateStruct.get(0);
+        final var tokenSymbol = (String) tokenCreateStruct.get(1);
+        final var tokenTreasury = addressIdConverter.convert(tokenCreateStruct.get(2));
+        final var memo = (String) tokenCreateStruct.get(3);
+        final var isSupplyTypeFinite = (Boolean) tokenCreateStruct.get(4);
+        final var maxSupply = (long) tokenCreateStruct.get(5);
+        final var isFreezeDefault = (Boolean) tokenCreateStruct.get(6);
+        final var tokenKeys = decodeTokenKeys(tokenCreateStruct.get(7), addressIdConverter);
+        final var tokenExpiry = decodeTokenExpiry(tokenCreateStruct.get(8), addressIdConverter);
```

Comment:
Please define those 0, 1, 2, .. etc as constants instead of the magic numbers in this class.